src/Deserialization/Transaction.purs

```diff
@@ -275,32 +306,139 @@ convertTxBody txBody = do
+convertUpdate :: forall (r :: Row Type). Csl.Update -> Err r T.Update
 convertUpdate u = do
   let { epoch: e, paramUpdates } = _unpackUpdate containerHelper u
-  epoch <- map Epoch $ cslNumberToUInt "convertUpdate: epoch" e
-  ppus <- traverse (traverse convertProtocolParamUpdate) paramUpdates
+  epoch <- map T.Epoch $ cslNumberToUInt "convertUpdate: epoch" e
+  ppus <- traverse
+    ( bitraverse
+        (pure <<< T.GenesisHash <<< toBytes <<< asOneOf)
+        convertProtocolParamUpdate
+    )
+    paramUpdates
   pure
     { epoch
-    , proposedProtocolParameterUpdates: ProposedProtocolParameterUpdates $
+    , proposedProtocolParameterUpdates: T.ProposedProtocolParameterUpdates $
         M.fromFoldable ppus
     }
 
 -- | TODO partially implemented
 convertCertificate
-  :: forall (r :: Row Type). Csl.Certificate -> Err r Certificate
+  :: forall (r :: Row Type). Csl.Certificate -> Err r T.Certificate
 convertCertificate = _convertCert certConvHelper
   where
   certConvHelper :: CertConvHelper r
   certConvHelper =
-    { stakeDeregistration: pure <<< StakeDeregistration
-    , stakeRegistration: pure <<< StakeRegistration
-    , stakeDelegation: \sc -> pure <<< StakeDelegation sc
-    , notImplementedError: notImplementedError
+    { stakeDeregistration: pure <<< T.StakeDeregistration
+    , stakeRegistration: pure <<< T.StakeRegistration
+    , stakeDelegation: \sc -> pure <<< T.StakeDelegation sc
+    , poolRegistration: convertPoolRegistration
+    , poolRetirement: convertPoolRetirement
+    , genesisKeyDelegation: \genesisHash genesisDelegateHash vrfKeyhash -> do
+        pure $ T.GenesisKeyDelegation
+          { genesisHash: T.GenesisHash $ toBytes $ asOneOf genesisHash
+          , genesisDelegateHash: T.GenesisDelegateHash
+              (toBytes $ asOneOf genesisDelegateHash)
+          , vrfKeyhash: vrfKeyhash
+          }
+    , moveInstantaneousRewardsToOtherPotCert: \pot amount -> do
+        pure $ T.MoveInstantaneousRewardsCert $
+          T.ToOtherPot { pot, amount: amount }
+    , moveInstantaneousRewardsToStakeCreds: \pot amounts -> do
+        pure $ T.MoveInstantaneousRewardsCert $
+          T.ToStakeCreds { pot, amounts: convertMIRToStakeCredentials amounts }
+    }
+
+convertMIRToStakeCredentials
+  :: Csl.MIRToStakeCredentials -> T.MIRToStakeCredentials
+convertMIRToStakeCredentials =
+  T.MIRToStakeCredentials <<< M.fromFoldable <<< unpackMIRToStakeCredentials_
+    containerHelper
+
+convertPoolRegistration
+  :: forall (r :: Row Type)
+   . Csl.PoolParams
+  -> Err r T.Certificate
+convertPoolRegistration params = do
+  relays <- traverse convertRelay $ poolParamsRelays containerHelper params
+  pure $ T.PoolRegistration
+    { operator: poolParamsOperator params
+    , vrfKeyhash: poolParamsVrfKeyhash params
+    , pledge: poolParamsPledge params
+    , cost: poolParamsCost params
+    , margin: _unpackUnitInterval $ poolParamsMargin params
+    , reward_account: poolParamsRewardAccount params
```

Comment:
Let's change this (and all other occurrences) to `rewardAccount`. Not sure if this got skipped during the great snake_case to camelCase migration, but if we don't change it now we're stuck with it until v2